src/api/mod.rs

```diff
@@ -1781,13 +1782,13 @@ impl<T: Pixel> ContextInner<T> {
     if cur_input_frameno == 0 {
       return;
     }
-    for i in 0..cur_input_frameno {
+    for i in self.frame_q.keys().next().cloned().unwrap_or(0)..cur_input_frameno {
```

Comment:
This change and 1791 aren't related to memory usage, but should provide a minor performance improvement especially on long videos.

Reply:
Please split them in a second patch.